src/clickmap_nav/tool_retrieval_interface.py

```diff
@@ -37,6 +48,12 @@ def __init__(self, robot, upload_path, model_path, silhouette=None, silhouetteAc
             'right_fisheye_image',
         ] # See conq/cameras_utils.py RGB_SOURCES
 
+        # Get locations to feed to open_ai_nav_client
+        self.waypoint_locations = [name for name in self.name_to_id.keys() if 'waypoint' not in name]
+
+        # Create Azure client
+        self.open_ai_nav_client = OpenAINavClient(locations=self.waypoint_locations)

```

Comment:
Not for this PR, but do you think it's worth it to create a base class for ChatGPT APIs so that the UMich and OpenAI APIs share the same interface?

It might be more trouble than it's worth so I wanted to ask you

Reply:
We could with a flag or something to designate which one to use, but the UM GPT client uses the `AzureOpenAI` class whereas if we go straight through OpenAI, we just use the `OpenAI `client. The UM GPT one also requires more input to the client, but the OpenAI one just uses a token.

Reply:
Got it. Seems like they're different enough that creating a base class wouldn't be terribly beneficial